database/rrddim.c

```diff
@@ -522,7 +522,7 @@ int rrddim_unhide(RRDSET *st, const char *id) {
 
     rrddim_flag_clear(rd, RRDDIM_FLAG_HIDDEN);
 #ifdef ENABLE_ACLK
-    aclk_update_chart(st->rrdhost, st->id);
+    aclk_update_chart(st->rrdhost, st->id, ACLK_CMD_CHART);
```

Comment:
same applies here i would prefer not to add ACLK as dependency for RRDDIM (there is many places like that). But as I said before i can accept this PR in this way (just prefer it to be done differently)

Reply:
Do we don't have callbacks unfortunately, which I would register and have it automatically fire. 
I also hated adding an extra param to the aclk_update_chart (tried with del_chart) but codacy started complaining about duplicate code.

Reply:
Why is `st->rrdhost` being passed to the aclk? (Considering the discussion about leaking pointers)

Reply:
I think the check against `localhost` and extraction of the hostname should happen in `rrddim.c` instead of in the aclk code.

Reply:
In any case though, this pattern is everywhere in the code. Meaning that you get `st->rrdhost` or `rd->rrdset` or `rd->rrdset->rrdhost` all the time.

At this point, it should be safe to assume that the pointer of a parent structure is not going away as long as you keep the the child structure alive.

Nobody can guarantee that the parent structure is not being modified though.

Reply:
I have tried both (passing hostname and *host). At this point there is no benefit either way and it needs to be revisited. In short I wanted a single line outside of non aclk code (eventually we may need to handle slaves as well if we reach a decision that slaves are automatically claimed)

Reply:
Ok, well it does not make the situation any worse than it already is.

Reply:
Using callbacks here does not remove the dependency between the codes - it moves it from being a compile-time dependency to a run-time dependency which has both benefits and drawbacks. The drawback is that it removes visibility of what the dependency is unless you go and find all the points in the code that add callbacks. In particular it breaks the ability of the editor or doxygen to give useful information.